mantle/kola/tests/misc/network.go

```diff
@@ -59,21 +60,30 @@ type listener struct {
 func checkListeners(c cluster.TestCluster, expectedListeners []listener) error {
 	m := c.Machines()[0]
 
-	output := c.MustSSH(m, "sudo netstat -plutn")
+	output := c.MustSSH(m, "sudo ss -plutn")
 
 	processes := strings.Split(string(output), "\n")
 	// verify header is as expected
 	if len(processes) < 2 {
-		c.Fatalf("expected at least two lines of nestat output: %q", output)
+		c.Fatalf("expected at least two lines of ss output: %q", output)
 	}
-	if processes[0] != "Active Internet connections (only servers)" {
-		c.Fatalf("netstat output has changed format: %q", output)
-	}
-	if !regexp.MustCompile(`Proto\s+Recv-Q\s+Send-Q\s+Local Address\s+Foreign Address\s+State\s+PID/Program name`).MatchString(processes[1]) {
-		c.Fatalf("netstat output has changed format: %q", output)
+	// ss output's header sometimes does not have whitespace between "Peer Address:Port" and "Process"
+	headerRegex := `Netid\s+State\s+Recv-Q\s+Send-Q\s+Local Address:Port\s+Peer Address:Port\s*Process`
+	if !regexp.MustCompile(headerRegex).MatchString(processes[0]) {
+		c.Fatalf("ss output has changed format: %q", processes[0])
 	}
 	// skip header
-	processes = processes[2:]
+	processes = processes[1:]
+
+	// create expectedListeners map
+	expectedListenersMap := map[listener]bool{}
+	for _, expected := range expectedListeners {
+		// allow expected tcp to match tcp6
+		expected.protocol = strings.TrimRightFunc(expected.protocol, func(r rune) bool {
+			return unicode.IsNumber(r)
+		})
+		expectedListenersMap[expected] = true
+	}
 
 NextProcess:
 	for _, line := range processes {
```

Comment:
might be nice to paste an example line of output somewhere in the code with trimmed whitespace (as much as possible)

Reply:
Was thinking the same since there are hard-coded indices through out the test and it might also help with future reference if the output format changes.

Reply:
Great suggestion, that would definitely help :)